bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/BaseThingHandler.java

```diff
@@ -509,7 +510,7 @@ protected void updateProperties(Map<String, String> properties) {
      * @param value the value of the property
      */
     protected void updateProperty(String name, String value) {
```

Comment:
Hm, ... shouldn't the `value` be annotated by `@Nullable` then?

Reply:
That makes sense and if we keep allowing null values the Map value type in `protected void updateProperties(Map<String, String> properties)` should also be properly annotated. I don't know how common it is for properties to be nullable. Another way could be to add a `removeProperty` method so we can rid of keys with `null` values. But `updateProperties` would also have to be updated for that and I don't know how breaking all these API changes would be. :wink: